handler.c

```diff
@@ -1281,6 +1294,51 @@ static EFI_STATUS verify_auth_var(uint8_t *name, UINTN name_len,
     return status;
 }
 
+static EFI_STATUS
+check_ppi_variables(uint8_t *name, UINTN name_len, EFI_GUID *guid,uint8_t *data, UINTN data_len)
+{
+    if (name_len == sizeof(TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME) &&
+            !memcmp(name,TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME, name_len) &&
+            !memcmp(guid, &gEfiTcg2PpiXenGuid, GUID_LEN)) {
+       if (data_len != sizeof (uint8_t)) {
+           DBG("Bad PPI lock write. size=%lu\n", data_len);
+           return EFI_INVALID_PARAMETER;
+       }
+       if (*data != 1) {
+           DBG("Bad PPI lock write. data=%hu\n", *data);
+           return EFI_INVALID_PARAMETER;
+       }
+       DBG("PPI lock set!\n");
+       return EFI_SUCCESS;
+    }
+
+    if ((name_len == sizeof(TCG2_PHYSICAL_PRESENCEFLAGS_NAME) &&
+            !memcmp(name,TCG2_PHYSICAL_PRESENCEFLAGS_NAME,name_len)) &&
+            !memcmp(guid, &gEfiTcg2PpiXenGuid, GUID_LEN)) {
+
+        EFI_STATUS status;
+        UINTN data_len;
+        uint8_t *data;
+
+        status = internal_get_variable(TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME,
+                                       sizeof(TCG2_PHYSICAL_PRESENCEFLAGSLOCK_NAME),
+                                       &gEfiTcg2PpiXenGuid, &data, &data_len);
+        if (status == EFI_SUCCESS) {
+
+            if (data_len != sizeof (uint8_t) ||
+                *data != 0) {
+                DBG("Attempt to set PPI flags while locked! Lock length %lu, value: %hu \n", data_len, data_len ? *data : 0xff);
+                return EFI_WRITE_PROTECTED;
+            }
+            free(data);
+        } else {
+           DBG("Attempt to set PPI flags while, but getting lock returned 0x%016lx\n", status);
+           return EFI_WRITE_PROTECTED;
```

Comment:
I think this should return `status`, i.e. the actual error. (It would likely be `EFI_DEVICE_ERROR` which is kind of like an internal error for UEFI.)